### PR TITLE
Parse POSIX time-zone strings using Combine

### PR DESCRIPTION
**Pull request by nordzilla**
Parses POSIX time-zone strings using Combine.

This is the first pull request to ease the burden of the review process and get early feedback.
The subsequent pull request will include the full TZIF parser which will use this parser.

### CODE REVIEW

File: experimental/tzif/Cargo.toml
Comment (sffc) [inline experimental/tzif/Cargo.toml:6]: Request: Moving forward, in general, all transformers / source providers should be defined within the `datagen` crate as a submodule under `transform`. The exception is if you make a general-purpose crate for parsing TZIF files, in which case you should name the crate something more general, and move it into the utils folder.

**Review by sffc — COMMENTED** [review]
I defer to @Manishearth for the detailed review of the posix parser.

File: experimental/tzif/Cargo.toml
Comment (Manishearth) [reply experimental/tzif/Cargo.toml:6]: I'd have utils/tzif and then stick the datagen stuff in experimental or under provider/datagen

but it's okay to use experimental now, up to you. experimental is anything goes :)

**Review by Manishearth — APPROVED** [review]
Parser looks very normal and good. I don't know too many specifics about the format but I think it's fine!

File: experimental/tzif/Cargo.toml
Comment (nordzilla) [reply experimental/tzif/Cargo.toml:6]: Great. @sffc 

Do you have a preference on the ordering here? I'm finishing up the 2nd PR for the TZif parser itself right now. 

Would you prefer to move this PR over to `utils` or land in experimental and then move it all after it's reviewed and landed?

Comment (Manishearth) [reply experimental/tzif/Cargo.toml:6]: I have no preference on ordering, whatever is most convenient for you. Shane might.

Comment (sffc) [reply experimental/tzif/Cargo.toml:6]: If in doubt, keep it in `experimental`.

Comment (sffc) [reply experimental/tzif/Cargo.toml:6]: Yeah, let's get all the code landed in `experimental` and then decide how to name it and where it should live.

**Pull request merged**
